app/controllers/clock_controller.rb

```diff
@@ -1,14 +1,27 @@
 class ClockController < ApplicationController
+  def main
+    current_timezone
+    @current_weather = current_weather(@current_timezone.name)
+  end
+
   def index
-    @current_timezone = Time.zone
+    current_timezone
+    @current_weather = current_weather(@current_timezone.name)
+    @user_timezones = current_user.timezones
   end
 
   def timezones
-    @timezones = ActiveSupport::TimeZone.all
+    @timezones = user_signed_in? ? all_timezones[1..5].map do |timezone|
+      {timezone: timezone, weather: current_weather(timezone.name)}
+    end : all_timezones.map { |timezone| {timezone: timezone} }
   end
 
-  def favorites
-    @user_timezones = current_user.timezones
+  def show
+    @timezone = get_new_support_timezone(params[:id])
+    @weather = []
+    weather_days = forecast_weather(@timezone.name)
+    days = weather_days["list"].length/5
+    5.times { |position| @weather.push(weather_days["list"][(position*days)+2]) }
```

Comment:
Here a service object would work also

Reply:
Yeah, I implemented it and it turned out cleaner